gix-index/src/entry/stat.rs

```diff
@@ -0,0 +1,415 @@
+        let stat1 = Stat {
+            mtime: Time { secs: 0, nsecs: 0 },
+            ctime: Time { secs: 0, nsecs: 0 },
+            dev: 0,
+            ino: 0,
+            uid: 0,
+            gid: 0,
+            size: 0,
+        };
+        let stat2 = Stat {
+            mtime: Time { secs: 0, nsecs: 0 },
+            ctime: Time { secs: 0, nsecs: 0 },
+            dev: 1,
+            ino: 0,
+            uid: 0,
+            gid: 0,
+            size: 0,
+        };
+
+        assert!(stat1.matches(&stat2, Options::default()));
+        assert!(!stat1.matches(
+            &stat2,
+            Options {
+                use_stdev: true,
+                ..Default::default()
+            }
+        ));
+    }
+
+    #[test]
+    fn check_stat() {
+        let stat1 = Stat {
+            mtime: Time { secs: 0, nsecs: 0 },
+            ctime: Time { secs: 0, nsecs: 0 },
+            dev: 0,
+            ino: 0,
+            uid: 0,
+            gid: 0,
+            size: 0,
+        };
+        let mut stat2 = stat1;
+        assert!(stat1.matches(&stat2, Options::default()));
+        assert!(stat1.matches(
+            &stat2,
+            Options {
+                check_stat: false,
+                ..Default::default()
+            }
+        ));
+        stat2.ino = 1;
+        assert!(!stat1.matches(&stat2, Options::default()));
+        assert!(stat1.matches(
+            &stat2,
+            Options {
+                check_stat: false,
+                ..Default::default()
+            }
+        ));
+        stat2 = stat1;
+        stat2.uid = 1;
+        assert!(!stat1.matches(&stat2, Options::default()));
+        assert!(stat1.matches(
+            &stat2,
+            Options {
+                check_stat: false,
+                ..Default::default()
+            }
+        ));
+        stat2 = stat1;
+        stat2.gid = 1;
+        assert!(!stat1.matches(&stat2, Options::default()));
+        assert!(stat1.matches(
+            &stat2,
+            Options {
+                check_stat: false,
+                ..Default::default()
+            }
+        ));
+        stat2 = stat1;
+        stat2.size = 1;
+        assert!(!stat1.matches(&stat2, Options::default()));
+        assert!(!stat1.matches(
+            &stat2,
+            Options {
+                check_stat: false,
+                ..Default::default()
+            }
+        ));
+    }
+
+    #[test]
+    fn racy_timestamp() {
+        let stat1 = Stat {
+            mtime: Time { secs: 1, nsecs: 10 },
+            ctime: Time { secs: 0, nsecs: 0 },
+            dev: 0,
+            ino: 0,
+            uid: 0,
+            gid: 0,
+            size: 0,
+        };
+        assert!(stat1.is_racy(FileTime::from_unix_time(1, 0), Options::default()));
+        assert!(stat1.is_racy(
+            FileTime::from_unix_time(1, 0),
+            Options {
+                use_nsec: true,
+                ..Default::default()
+            }
+        ));
```

Comment:
Here it would definitely be beneficial for quick understanding why it's racy or why it's not, just to aid understanding and speed things up for those new to the codebase.

Reply:
I added comments in [5edcc13](https://github.com/Byron/gitoxide/pull/805/commits/5edcc1399519611ec6380bb1ca6ad66072fa0b64).

This `is_racy` check really only does `mtime >= timestamp` (while respecting the `use_nsec` option).